### PR TITLE
refactor: Reorder intrinsics in sse2neon.h

### PR DESCRIPTION
**Pull request by marktwtn**
Reorder the intrinsics with the SSE, SSE2, SSE3, SSSE3, SSE4.1 and
SSE4.2 classes.

### CODE REVIEW

**Comment by jserv** [comment]
We need the guideline to add intrinsics as well.

**Comment by marktwtn** [comment]
> We need the guideline to add intrinsics as well.

I have added a simple guideline in `README.md`.

File: README.md [hunk 1]
```diff
@@ -71,6 +71,14 @@ Considering the balance between correctness and performance, `sse2neon` recogniz
 
 The above are turned off by default, and you should define the corresponding macro(s) as `1` before including `sse2neon.h` if you need the precise implementations.
 
+## Add New Intrinsic
```
Comment (jserv) [inline README.md:74]: Move this sub-section to the place between "Run Built-in Test suite" and "Coding Convention."

Comment (marktwtn) [reply README.md]: Done.

**Pull request merged**